src/internal/task/task_threads.c

```diff
@@ -22,7 +30,11 @@ struct state_pass {
     void      *args;
     void      *task;
     uintptr_t *stackTop;
+    #if __APPLE__
```

Comment:
Preprocessor directives are somewhat ugly, and get uglier with the addition of platforms (e.g. Windows). I suggest adding a GOOS-specific typedef for the semaphore type along with a handful of small inline functions that abstracts the semaphore API.

Reply:
That is what I originally tried, but in my opinion didn't result in code that was easier to read. Note that the way these functions are called is very different between standard POSIX (`sem_*` API) and MacOS.

I will likely make a new file for Windows. Maybe I can use the pthread library of MinGW and reuse this code, but I expect Windows will be so different a separate file makes things easier. We'll see.